fdbclient/S3BlobStore.actor.cpp

```diff
@@ -302,10 +302,11 @@ Reference<S3BlobStoreEndpoint> S3BlobStoreEndpoint::fromString(const std::string
 			}
 
 			// The parameter is known to S3BlobStoreEndpoint so it must be numeric and valid.
-			char* valueEnd;
-			int ivalue = strtol(value.toString().c_str(), &valueEnd, 10);
-			if (*valueEnd || (ivalue == 0 && value.toString() != "0"))
-				throw format("%s is not a valid value for %s", value.toString().c_str(), name.toString().c_str());
+			char* valueEnd = nullptr;
+			std::string s = value.toString();
+			int ivalue = strtol(s.c_str(), &valueEnd, 10);
+			if (*valueEnd || (ivalue == 0 && s != "0"))
```

Comment:
Probably this change has been reviewed already, but according to "strtol" documentation "(value == 0)" itself indicates no valid conversion could be found so why further checking for (s != "0")?

Reply:
"0" is a valid string itself

Reply:
Makes sense.

Not sure what happened to my other comment here: why are we not checking for "(ivalue == LONG_MAX)" (or LONG_MIN)? And, why "ivalue" is not of type "long int"?

Reply:
Good point. I fixed return type and added checking.